controllers/agents/app/servicebinding_controller.go

```diff
@@ -555,3 +595,132 @@ func (r *ServiceBindingReconciler) unbindApplications(ctx context.Context, req c
 	}
 	return nil
 }
+
+func (r *ServiceBindingReconciler) SetWatchersForResources(ctx context.Context, serviceBinding v1alpha1.ServiceBinding) error {
+	reconcileLog := log.FromContext(ctx)
+	typemeta := metav1.TypeMeta{
+		Kind:       serviceBinding.Spec.Application.Kind,
+		APIVersion: serviceBinding.Spec.Application.APIVersion,
+	}
+	gvk := typemeta.GroupVersionKind()
+	mapping, err := r.Client.RESTMapper().RESTMapping(gvk.GroupKind(), gvk.Version)
+	if err != nil {
+		reconcileLog.Error(err, "error on creating mapping")
+		return err
+	}
+	reconcileLog.Info("resource to be watched", "resource", mapping.Resource)
+	if err = r.RunInformer(ctx, mapping.Resource, serviceBinding); err != nil {
+		reconcileLog.Error(err, "error running informer")
+		return err
+	}
+	return nil
+}
+
+func (r *ServiceBindingReconciler) RunInformer(ctx context.Context, resource schema.GroupVersionResource, serviceBinding v1alpha1.ServiceBinding) error {
+	l := log.FromContext(ctx)
+
+	// check if informer already exists
+	if _, ok := r.informers[serviceBinding.GetName()]; ok {
+		l.Info("Informer already exists")
+		return nil
+	}
+	clusterConfig, err := rest.InClusterConfig()
+	if err != nil {
+		l.Info("failed creating cluster config")
+		panic(err)
+	}
+	clusterClient, err := dynamic.NewForConfig(clusterConfig)
+	if err != nil {
+		l.Info("failed creating cluster config")
+		panic(err)
+	}
+	factory := dynamicinformer.NewFilteredDynamicSharedInformerFactory(clusterClient, time.Minute, serviceBinding.Namespace, nil)
+	i := factory.ForResource(resource).Informer()
+
+	var synced atomic.Bool
+	synced.Store(false)
+	if _, err := i.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			if !synced.Load() {
+				return
+			}
+			var psSecret v1.Secret
+			var applicationResourceList []unstructured.Unstructured
+			applicationResource := obj.(*unstructured.Unstructured)
+			if applicationResource.GetName() == constants.ApplicationAgentDeploymentName {
+				return
+			}
+			applicationResourceList = append(applicationResourceList, *applicationResource)
+			l.Info(fmt.Sprintf("application resource %v", applicationResourceList))
+			if psSecret, err = r.GetSecret(ctx, serviceBinding, applicationResourceList); err != nil {
+				return
+			}
+			if err := r.PrepareBinding(ctx, serviceBinding, applicationResourceList, &psSecret); err != nil {
+				return
+			}
```

Comment:
should we retry, or log the error or update status conditions? applies also below

Reply:
`AddEventHandler` methods is implementing interface `ResourceEventHandler ` which does not allow error to be returned.